docs/src/capi/explanation/in-place-upgrades.md

```diff
@@ -0,0 +1,237 @@
+```
+
+### Machine Upgrade Process
+
+{{product}} has a daemon running in the background called the `k8sd`. 
+It’s responsible for many things in the context of {{product}} but 
+one of them is to expose certain endpoints that can be used to 
+interact with the cluster. The single machine upgrader calls the 
+`/snap/refresh` endpoint on the machine that it’s trying to upgrade. 
+This endpoint call will trigger the “actual” upgrade process and in 
+the meantime, the `/snap/refresh-status` will be called periodically 
+by the single machine upgrader to see how things are going. It’s worth 
+noting that ensuring secure communication between the single machine 
+upgrader and the {{product}} daemon (k8sd) is really important and 
+something that is handled internally as well.
+
+```{figure} ../../assets/in-place-upgrades/k8sd-call.png
+   :width: 100%
+   :alt: Single Machine In-Place Upgrade Controller Calling The `k8sd` 
+   Endpoints
+
+   Single Machine In-Place Upgrade Controller Calling The `k8sd` Endpoints
+```
+
+### Orchestrated In-Place Upgrade Controller
+
+While the “Single Machine In-Place Upgrade Controller” is responsible 
+for upgrading individual machines, if our workload cluster is made up 
+of tens, hundreds, or thousands of nodes, annotating them one by one and 
+keeping track of their statuses, failures, reasons and errors can become 
+a daunting or even impossible task. “Orchestrated In-Place Upgrade 
+Controller” to the rescue!
+
+The main idea behind this class of controllers is to enable us upgrade 
+multiple machines by only annotating a single object: the owner of 
+those machines. In CAPI, we mostly have two main machine groups in 
+our workload cluster: “control-plane-node machines” and “worker-node 
+machines”.
+
+Let’s say we want to upgrade all of our worker nodes. In this case, we 
+only need to annotate the `MachineDeployment` object. The orchestrated 
+upgrade controller watches for the `upgrade-to` annotation on the 
+`MachineDeployment` and will trigger an in-place upgrade for all of 
+its owned machines by annotating them one by one. The responsibility 
+is then delegated to the single machine upgrader to make sure each 
+individual machine is getting upgraded successfully. The orchestrator 
+is only there to keep track of the upgrade status of these machines, 
+reported by the single machine upgrader (by annotating the machine). 
+If any of the machines fail to get upgraded, the orchestrator marks 
+the owner (here, `MachineDeployment`) with the respective annotations 
+(`status: failed` and `last-failed-attempt-at`), publishes helpful 
+and informative events and trusts the single machine controller to do 
+the retry and eventually succeed. When all the owned machines are 
+upgraded as expected, the orchestrator considers the operation successful, 
+marks the owner with `release` and `status: done` and steps down.
+
+To have a better understanding, let’s have a look the flow of the orchestrator:
```

Comment:
Maybe "To paint a better picture" or "For a clearer picture" or "To better appreciate this" ?